src/rules.ts

```diff
@@ -157,7 +213,9 @@ block.pedantic = {
 /**
  * Inline-Level Grammar
  */
-export const inline = {
+// Not all rules are defined in the object literal
+// @ts-expect-error
+export const inline: Record<InlineRuleNames, Rule> & Record<InlineSubRuleNames, Rules> & Rules = {
```

Comment:
If we use `as const` at the end of `inline`, instead of using types here, would that solve the problem below so we could remove `as Rule`?

Reply:
No because it would then complain for each `inline.additionalRule = ...` line with `TS2551: Property  _punctuation  does not exist on type` or `TS2540: Cannot assign to  punctuation  because it is a read-only property.`